src/CBF/CBF.jl

```diff
@@ -0,0 +1,588 @@
+module CBF
+
+using MathOptInterface
+
+const MOI = MathOptInterface
+const MOIU = MOI.Utilities
+
+MOIU.@model(InnerCBFModel,
+    (MOI.Integer,),
+    (),
+    (MOI.Reals, MOI.Zeros, MOI.Nonnegatives, MOI.Nonpositives,
+        MOI.SecondOrderCone, MOI.RotatedSecondOrderCone,
+        MOI.PositiveSemidefiniteConeTriangle,
+        MOI.ExponentialCone, MOI.DualExponentialCone),
+    (MOI.PowerCone, MOI.DualPowerCone),
+    (MOI.SingleVariable,),
+    (MOI.ScalarAffineFunction,),
+    (MOI.VectorOfVariables,),
+    (MOI.VectorAffineFunction,)
+)
+
+const Model = InnerCBFModel{Float64}
+
+function Base.show(io::IO, ::Model)
+    print(io, "A Conic Benchmark Format (CBF) model")
+    return
+end
+
+# ==============================================================================
+#
+#   MOI.write_to_file
+#
+# ==============================================================================
+
+function MOI.write_to_file(model::Model, filename::String)
+    open(filename, "w") do io
+        println(io, "# ", MOI.get(model, MOI.Name())) # Name into CBF comments.
+        println(io)
+
+        println(io, "VER") # CBF version number.
+        println(io, 3)
+        println(io)
+
+        # Objective sense.
+        println(io, "OBJSENSE")
+        objsense = MOI.get(model, MOI.ObjectiveSense())
+        if objsense == MOI.MaxSense
+            println(io, "MAX")
+        else
+            println(io, "MIN") # Includes the case of MOI.FeasibilitySense.
+        end
+        println(io)
+
+        # Variables.
+        nvar = MOI.get(model, MOI.NumberOfVariables())
+        println(io, "VAR")
+        println(io, nvar, " 1")
+        println(io, "F ", nvar)
+        println(io)
+
+        # Helper functions for MOI.
+        getmodelcons(F, S) = MOI.get(model, MOI.ListOfConstraintIndices{F, S}())
+        getconfun(conidx) = MOI.get(model, MOI.ConstraintFunction(), conidx)
+        getconset(conidx) = MOI.get(model, MOI.ConstraintSet(), conidx)
```

Comment:
Ditto with naming. `model_constraints(F, S)`, `constraint_function`, `constraint_set(con_idx)`. We don't mind verbosity.